### PR TITLE
Remove parallel for from data table to fix computed column threading issue in #1831

### PR DESCRIPTION
**Pull request by timkpaine**

### CODE REVIEW

File: cpp/perspective/src/include/perspective/gnode.h [hunk 1]
```diff
@@ -208,7 +208,7 @@ class PERSPECTIVE_EXPORT t_gnode {
     std::shared_ptr<t_expression_vocab> get_expression_vocab() const;
     std::shared_ptr<t_regex_mapping> get_expression_regex_mapping() const;
 
-#ifdef PSP_PARALLEL_FOR
+#ifdef PSP_ENABLE_PYTHON
```
Comment (timkpaine) [inline cpp/perspective/src/include/perspective/gnode.h:211]: this is unrelated but should be enabled conditional on PYTHON, not PARALLEL_FOR

**Comment by timkpaine** [comment]
@texodus might need your help benchmarking this, if it causes a substantial degradation in performance I can try to break this loop apart to reenable some part in parallel

**Review by texodus — CHANGES_REQUESTED** [review]
It's not clear to my why this change would impact the test provided, so on a hunch I forked just this test to a new branch and it [passes just the same on all Linux builds](https://dev.azure.com/finosfoundation/perspective/_build/results?buildId=4456&view=logs&j=9c1f2f8e-4d43-59a0-c5f5-4a00727b21e2&t=19e3a096-6506-57cc-343b-951c3932e971&l=1558).  What am I missing?

**Comment by timkpaine** [comment]
This test fails on my Linux machine, validated on 2 other Linux machines

**Comment by timkpaine** [comment]
See #1835 for more details

**Comment by timkpaine** [comment]
My guess would be either azure or the manylinux images themselves don't trigger the test case to fail.

**Comment by timkpaine** [comment]
Some additional testing suggests this might've been fixed already via the arrow upgrade https://github.com/finos/perspective/pull/1816

**Comment by texodus** [comment]
So thanks to the accompanying test (which I _do_ want to merge), I can confirm that this issue was resolved by #1793, and is unrelated to Arrow/threading.

**Pull request closed without merging**